WSLBlobNFS/WSLBlobNFS.psm1

```diff
@@ -6,178 +11,539 @@ Set-PSDebug -Strict
+}
+
+function Execute-WSL
+{
+    [CmdletBinding()]
+    param(
+        [Parameter(Mandatory = $true)][string]$wslcommand
+    )
+
+    Write-Debug "Executing $wslcommand"
+    wsl -d $distroName -u $userName -e bash -c $wslcommand
+}
+
+function Copy-WSLFiles
+{
+    [CmdletBinding()]
+    param()
+
+    # Note: Quote the path with '' to preserve space
+    Write-Verbose "Copying necessary linux files from $modulePathForLinux"
+
+    Execute-WSL "mkdir -p $linuxScriptsPath"
+
+    Execute-WSL "cp '$modulePathForLinux/$wslScriptName' $linuxScriptsPath/$wslScriptName"
+
+    # Files saved from windows will have \r\n line endings. Hence, we need to remove \r.
+    Execute-WSL "sed -i -e 's/\r$//' $linuxScriptsPath/$wslScriptName"
+
+    Execute-WSL "chmod +x $linuxScriptsPath/$wslScriptName"
+
+    Execute-WSL "cp '$modulePathForLinux/$queryScriptName' $linuxScriptsPath/$queryScriptName"
+
+    # Files saved from windows will have \r\n line endings. Hence, we need to remove \r.
+    Execute-WSL "sed -i -e 's/\r$//' $linuxScriptsPath/$queryScriptName"
+
+    Execute-WSL "chmod +x $linuxScriptsPath/$queryScriptName"
+
+    Execute-WSL "ls -l $linuxScriptsPath | grep -E '$wslScriptName|$queryScriptName' > /dev/null 2>&1"
+}
+
+function Intall-WSLBlobNFS-Internal
+{
+    [CmdletBinding()]
+    param()
+
     # Check if wsl is installed or not.
-    $wslstatus = wsl -l -v
-    if($LASTEXITCODE -eq 1)
+    wsl -l -v > $null 2>&1
+    $wslstatus = $LASTEXITCODE
+
+    # Check if the distro is installed or not.
+    # Note: executing a command such as ls is required else if the distro is installed, then control will not come out of wsl command.
+    wsl -d $distroName ls > $null 2>&1
+    $distrostatus = $LASTEXITCODE
+
+    if($wslstatus -eq 1)
     {
         Write-Host "WSL is not installed. Installing WSL..."
 
-        wsl --install -d Ubuntu-22.04
-        Write-Host "Restart the VM to complete WSL installation, and then Run the script again with Initialize-WSLBlobNFS to continue the WSL setup."
+        wsl --install -d $distroName
+        # To-do: Check if the distro is installed successfully or not.
+
+        # To-do: Prompt user to restart the machine to complete WSL installation.
+        Write-Success "Restart the VM to complete WSL installation, and then run Initialize-WSLBlobNFS to continue the WSL setup."
+
+        # Set the exit code to 1 to indicate that the script execution is not completed.
+        $global:LASTEXITCODE = 1
         return
     }
-    Write-Host "WSL is already installed. Skipping WSL installation."
+    elseif($distrostatus -eq -1)
+    {
+        Write-Host "WSL distro $distroName is not installed. Installing WSL distro $distroName..."
+        wsl --install -d $distroName
+        # To-do: Check if the distro is installed successfully or not.
+
+        Write-Success "WSL distro $distroName is installed. Run Initialize-WSLBlobNFS to setup WSL environment for WSLBlobNFS usage."
```

Comment:
nit,
again "WSL distro $distroName successfully installed!" in one line, followed by the rest will be more comforting to the user